### PR TITLE
[move-stdlib] Port fixedpoint32, compare, hash, bcs tests to unit tests, update bitvector tests

### PR DESCRIPTION
**Pull request by tzakian**
This PR removes a number of non-unit tests that should be unit tests for the stdlib, and ports them over to use Move unit tests.

### CODE REVIEW

File: language/move-stdlib/tests/BCSTests.move [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    #[test]
+    fun bcs_u128() {
+        let expected_output = x"01000000000000000000000000000000";
+        assert(BCS::to_bytes(&1u128) == expected_output, 0);
+    }
+
+    #[test]
+    fun bcs_vec_u8() {
```
Comment (tnowacki) [inline language/move-stdlib/tests/BCSTests.move:46]: could probably use some more tests for simple structs and vectors but that can be another PR

**Review by tnowacki — APPROVED** [review]

File: language/move-stdlib/tests/BCSTests.move (hunk 1 above)
Comment (tzakian) [reply language/move-stdlib/tests/BCSTests.move:46]: Agreed. This is just a pretty straightforward port. But there are definitely other tests we'll want to add -- I'll probably add those in next week sometime.
